### PR TITLE
FRONT-60: Detail page

### PR DESCRIPTION
**Pull request by DamianGrosz2**
## Description of Issue

Code the detail page, which includes several elements such as a sidebar on the right and left, a navigation bar at the top, project information, a diagram for stars, and cards for tweets/community sentiment.

## How I implemented

The main elements of the page have been broken down into individual components to promote reusability and maintainability. Here's a brief overview of the component structure:

- **Left Sidebar**: This sidebar currently has two entries. A saved searches tab is also included in the sidebar.
- **Navigation Bar**: A navigation bar is present at the top of the page.
- **Project Information**: Project-specific information is displayed in a dedicated component.
- **Diagram**: A diagram component has been built to visually represent various data points such as stars, forks, and issues. This component utilizes the recharts library for creating interactive and responsive charts.
- **Cards** for Tweets/Community Sentiment: These cards display data related to the social sentiments towards the project.
- **Right Sidebar**: This sidebar houses various project statistics. Each statistic is encapsulated within its own component for better separation of concerns.
- **Loading page**: The loading page (/loading) as in Figma with a progress bar, after its loaded it redirects to the overview page
- For more information on the reusable components that were created, please refer to the documentation in [Notion](https://www.notion.so/Reusable-Frontend-components-82b6640878b2490499d508adf57b0560?pvs=4).

## Other

The overview page bundles all of those components for now. It is not yet ion a clean code manner.
Also the Chart component can be improved. Both of those files need additional context (how does the data look like etc.).

Libraries:
- date-fns is needed to get the format for the date for the chart.
- primer/octicons-react: This is a library for the official GitHub Icons

Please review the PR and provide your feedback.

### CODE REVIEW

**Review by cnk1dev — CHANGES_REQUESTED** [review]
Components generally look good, a few style issues can be improved later on, but the following things need to be done before merging:

- [ ] Fix all eslint errors (200+ problems, many of them might be solvable with --fix option)
- [x] Make sure the build runs successfully (check out the Vercel deployment logs)
- [x] Remove unused icons folder

File: package.json [hunk 1]
```diff
@@ -15,14 +15,20 @@
...
     "@types/react": "18.2.0",
     "@types/react-dom": "18.2.1",
     "autoprefixer": "10.4.14",
+    "date-fns": "^2.30.0",
+    "feather": "^0.0.5",
     "next": "13.3.1",
     "postcss": "8.4.23",
     "react": "18.2.0",
     "react-dom": "18.2.0",
+    "react-feather": "^2.0.10",
```
Comment (pczern) [inline package.json:29]: I think this package should be removed, as `react-icons` includes it

File: src/components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      </>
+    ) : (
+      <>
+        {Icon && <Icon className={`mr-2 h-4 w-4 ${iconColor}`} />}
+        {text && <span className={`text-14 ${textColor}`}>{text}</span>}
+      </>
+    )
+  }, [Icon, text, switchOrder, iconColor, textColor])
+
+  let buttonStyle = ''
```
Comment (pczern) [inline src/components/Button.tsx:48]: use a Map

Comment (pczern) [reply src/components/Button.tsx:48]: see PR from Arno

File: src/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+const Button = ({
+  variant,
+  onClick,
+  text,
+  Icon,
+  switchOrder,
+  iconColor = 'text-icon-color',
+  textColor = 'text-text-secondary',
```
Comment (pczern) [inline src/components/Button.tsx:28]: who created these classes `text-text-secondary`? It is meant to be called `text-secondary`

Comment (pczern) [reply src/components/Button.tsx:28]: Speak with Arno about how to fix the color scheming, he has the same thing going on, I think he will fix it and then we can merge it with this PR

File: src/components/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      onClick={onClick}
+      className={`flex items-center rounded-[5px] transition-colors duration-100 hover:bg-bg-secondary-highlighted ${buttonStyle}`}
+    >
+      {contentLayout}
+    </button>
+  )
+}
+
+Button.defaultProps = {
+  text: '',
```
Comment (pczern) [inline src/components/Button.tsx:89]: use `null`

File: src/components/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  Icon: undefined,
```
Comment (pczern) [inline src/components/Button.tsx:90]: use `null`

File: src/components/Button.tsx (hunk 4 above)
Comment (pczern) [reply src/components/Button.tsx:89]: or remove it from defaultProps

File: src/components/Button.tsx (hunk 5 above)
Comment (pczern) [reply src/components/Button.tsx:90]: props not passed are undefined

**Review by cnk1dev — CHANGES_REQUESTED** [review]
Please resolve merge conflicts in tailwind config and global styles

File: src/components/DetailPage/Card.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import { ComponentType, ReactNode } from 'react'
+
+type CardProps = {
+  Icon: ComponentType<{ className?: string }>
```
Comment (pczern) [inline src/components/DetailPage/Card.tsx:4]: I think JSX.Element might be better or JSXElementConstructor see https://react-typescript-cheatsheet.netlify.app/docs/advanced/types_react_api/#typesreact

File: src/components/DetailPage/Card.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  name: string
+  button: ReactNode
+  textFields: string[]
+}
+
+const Card = ({ Icon, name, button, textFields }: CardProps) => (
+  <div className="mx-7 my-4 rounded-lg border border-border-color bg-bg-secondary">
+    <div className="flex flex-row items-center px-4 pt-4">
+      <Icon className="mr-2 text-highlight-color" />
+      <h2 className="">{name}</h2>
```
Comment (pczern) [inline src/components/DetailPage/Card.tsx:14]: useless className prop

File: tailwind.config.js [hunk 8]
```diff
@@ -9,8 +9,34 @@ module.exports = {
     extend: {
       backgroundImage: {
         'gradient-radial': 'radial-gradient(var(--tw-gradient-stops))',
-        'gradient-conic': 'conic-gradient(from 180deg at 50% 50%, var(--tw-gradient-stops))'
-      }
+        'gradient-conic': 'conic-gradient(from 180deg at 50% 50%, var(--tw-gradient-stops))',
+        'radial-gradient': 'radial-gradient(circle at 50% -70%, #191D3B, #191D3B, #191A23)',
+      },
+      colors: {
+        'bg-primary': '#191A23',
```
Comment (pczern) [inline tailwind.config.js:16]: speak with Arno about colors, but having `bg-bg-...` classes is not the way to go

File: package.json [hunk 9]
```diff
@@ -15,14 +15,20 @@
     "react:update-react-imports": "npx react-codemod update-react-imports"
   },
   "dependencies": {
+    "@primer/octicons-react": "^19.1.0",
     "@types/node": "18.16.3",
     "@types/react": "18.2.0",
     "@types/react-dom": "18.2.1",
     "autoprefixer": "10.4.14",
+    "date-fns": "^2.30.0",
```
Comment (pczern) [inline package.json:23]: I need an explanation of the different libraries in the PR

**Review by pczern — CHANGES_REQUESTED** [review]
Very good work, some updates are needed
More explanation about tech choices, libraries,
Please write me on Slack when changes are implemented.
Also please smaller PRs with at most 15 files.

File: src/components/Button.tsx (hunk 3 above)
Comment (ArnoClaude) [reply src/components/Button.tsx:28]: @DamianGrosz2 I've added color codes to the main branch. I've used the colors from the [Linear](https://www.figma.com/file/8YtjvVMikUG5aBl73t7xFz/Linear-Design-System-(Community)?node-id=8-2&t=6r3YUXN1Xo3eTOxG-0) design system, as well as the official [Tailwind](https://tailwindcss.com/docs/customizing-colors) colors for Gray 50 - 950 and Indigo 50 - 950.

Comment (ArnoClaude) [reply src/components/Button.tsx:28]: Lmk if those will be enough colors for our app (I think yes). And then, let's use these color codes, such that we can keep a consistent style and naming going onward :)

**Review by pczern — CHANGES_REQUESTED** [review]
Too many files, PR too big
Resolve merge conflicts, make sure tests pass
Then PR get another review

File: src/pages/DetailPage/ProjectInformation.tsx [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+            {text}
+          </p>
+        ))}
+      </div>
+      <Button
+        variant="highlighted"
+        text="Bookmark"
+        Icon={StarIcon}
+        onClick={() => {
+          console.log('clicked')
```
Comment (cnk1dev) [inline src/pages/DetailPage/ProjectInformation.tsx:29]:
```suggestion
```

File: src/pages/loading.tsx [hunk 11]
```diff
@@ -0,0 +1,59 @@
+/* eslint-disable no-console */
```
Comment (cnk1dev) [inline src/pages/loading.tsx:1]:
```suggestion
```

File: src/pages/loading.tsx [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      setProgress((oldProgress) => {
+        if (oldProgress === 100) {
+          clearInterval(timer)
+          navigate().then(
+            () => {
+              console.log('Navigation successful!')
```
Comment (cnk1dev) [inline src/pages/loading.tsx:22]: Remove console.log statements

File: src/pages/overview.tsx [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+import Chart from '@/pages/DetailPage/Chart'
+import InformationRow from '@/components/InformationRow'
+import ScatterPlot from '@/pages/DetailPage/ScatterPlot'
+import LeftSidebar from '@/components/LeftSidebar'
+
+const inter = Inter({ subsets: ['latin'] })
+
+const Overview = () => {
+  const handleClick = () => {
+    console.log('Menu item clicked')
```
Comment (cnk1dev) [inline src/pages/overview.tsx:37]:
```suggestion
    // Menu item clicked
```

**Review by cnk1dev — CHANGES_REQUESTED** [review]
Please also address unresolved comments from @pczern and run build locally to fix errors

File: src/pages/overview.tsx [hunk 14]
```diff
@@ -0,0 +1,261 @@
...
+                  }
+                  textFields={[
+                    'Lorem1 ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore',
+                    'Lorem2 ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore',
+                    'Lorem 3ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore'
+                  ]}
+                />
+                <Card
+                  Icon={FaHackerNews}
+                  name="Community sentiment"
```
[the remark this answers is not in the document]
Comment (cnk1dev) [reply src/pages/overview.tsx:140]: @ArnoClaude it's not top hackernews posts though, it is indeed supposed to be a summary of the community sentiment

**Comment by ArnoClaude** [comment]
image

Let's put the buttons above the chart, and we can figure out how to use the empty space on the right later. Right now, the buttons look awkwardly placed.
Also, remove the "value" indication, it doesn't provide any value (badum-tss).
And for the date, smth. like "Apr 2023" would be more readable.

**Review by cnk1dev — APPROVED** [review]

File: src/components/DetailPage/ProjectInformation.tsx [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+const ProjectInformation = ({ name, eli5, tags }: ProjectInformationProps) => (
+  <div className="border-b border-gray-800 px-7 py-4">
+    <div className="mb-4 flex flex-row items-center justify-between">
+      <div className="flex flex-row items-center">
+        <h1 className="mr-3">{name}</h1>
+        {tags.map((text) => (
+          <p
+            key={text}
```
Comment (pczern) [inline src/components/DetailPage/ProjectInformation.tsx:18]: text is not a key

File: src/pages/loading.tsx [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+
+    return () => {
+      clearInterval(timer)
+    }
+  })
+
+  return (
+    <main className={`${inter.className} flex min-h-screen flex-col`}>
+      <div className="flex grow flex-col items-center justify-between bg-radial-gradient">
+        <div />
```
Comment (pczern) [inline src/pages/loading.tsx:33]: the div's are missing the keys

File: src/components/RightSidebar/SocialMediaItem.tsx [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+  onClick: () => void
+}
+const SocialMediaItem = ({ Icon, text, name, value, growth, onClick }: SocialMediaProps) => (
+  <div className="flex flex-col justify-between">
+    <div className="inline-flex px-7 py-2.5 transition-colors duration-100 hover:bg-gray-850">
+      <div className="flex flex-row items-center justify-center gap-[15px]">
+        <button type="button" onClick={onClick}>
+          <Icon className="h-[14px] w-[14px] text-gray-500" />
+        </button>
+        <span className="text-xs not-italic leading-3 text-gray-500">{name + text}</span>
```
Comment (pczern) [inline src/components/RightSidebar/SocialMediaItem.tsx:18]: something not right here, there is no space or anything, could be one prop

**Review by pczern — COMMENTED** [review]
New review

File: package.json [hunk 18]
```diff
@@ -16,11 +16,13 @@
     "react:update-react-imports": "npx react-codemod update-react-imports"
   },
   "dependencies": {
+    "@primer/octicons-react": "^19.1.0",
```
[the remark this answers is not in the document]
Comment (pczern) [reply package.json:19]: they have a PR for @primer octicons but it's not merged yet

[the remark this answers is not in the document]
Comment (pczern) [reply package.json:19]: will use this package
